esrally/driver/driver.py

```diff
@@ -587,7 +587,7 @@ def receiveMsg_StartLoadGenerator(self, msg, sender):
         self.client_id = msg.client_id
         self.config = load_local_config(msg.config)
         self.abort_on_error = self.config.opts("driver", "on.error") == "abort"
-        self.es = client.EsClientFactory(self.config.opts("client", "hosts"), self.config.opts("client", "options")).create()
+        self.es = client.EsClientsFactory()(self.config.opts("client", "hosts").all_hosts, self.config.opts("client", "options").all_client_options)
```

Comment:
That syntax looks odd to me (constructor call followed by a `__call__` method invocation). Maybe we don't need a class for this and just use a function instead?

Reply:
Will replace with a function.